packages/wp-story-editor/src/components/fontCheck/fontCheckDialog.js

```diff
@@ -0,0 +1,177 @@
+  }));
+
+  const onSettingsClick = useCallback((evt) => {
+    trackClick(evt, 'click_font_settings');
+  }, []);
+
+  const updateMissingFontWithDefault = useCallback(() => {
+    updateElementsByFontFamily({
+      family: missingFont,
+      properties: { font: TEXT_ELEMENT_DEFAULT_FONT },
+    });
+    closeDialog();
+  }, [closeDialog, missingFont, updateElementsByFontFamily]);
+
+  const updateMissingFontWithSelected = useCallback(() => {
+    updateElementsByFontFamily({
+      family: missingFont,
+      properties: { font: suggestedFont },
+    });
+    closeDialog();
+  }, [suggestedFont, closeDialog, missingFont, updateElementsByFontFamily]);
+
+  const Chevron = isRTL ? ChevronRight : ChevronLeft;
+
+  return (
+    <Dialog
+      isOpen={isOpen}
+      onClose={defaultCloseAction}
+      title={__('Missing Fonts', 'web-stories')}
+      contentLabel={__('Missing Fonts', 'web-stories')}
+      actions={
+        <>
+          <Link href={dashboardLink}>
+            <DashboardLink>
+              <Chevron />
+              {__('Back to dashboard', 'web-stories')}
+            </DashboardLink>
+          </Link>
+          <Button
+            type={BUTTON_TYPES.SECONDARY}
+            size={BUTTON_SIZES.SMALL}
+            onClick={updateMissingFontWithDefault}
+          >
+            {__('Open anyway', 'web-stories')}
+          </Button>
+          <Button
+            type={BUTTON_TYPES.PRIMARY}
+            size={BUTTON_SIZES.SMALL}
+            onClick={updateMissingFontWithSelected}
+          >
+            {__('Replace font', 'web-stories')}
+          </Button>
+        </>
+      }
+    >
+      <DialogContent>
+        <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+          {__(
+            'This story contains the following font that is no longer available:',
+            'web-stories'
+          )}
+        </Text>
+        <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL} isBold>
+          {missingFont}
+        </Text>
+        <div>
+          <TranslateWithMarkup
+            mapping={{
+              a: (
+                <Link
+                  size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}
+                  onClick={onSettingsClick}
+                  href={dashboardSettingsLink}
+                  isBold
+                />
+              ),
+            }}
+          >
+            {__(
+              'To replace this font with your own or re-add it, go to <a>Settings</a>. To replace this font with a default, open the story.',
+              'web-stories'
+            )}
+          </TranslateWithMarkup>
+        </div>
+        <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+          {__('Or, choose replacement font below:', 'web-stories')}
+        </Text>
+        <FontPicker
+          zIndex={10}
+          currentValue={suggestedFont.family}
+          onChange={(font) => setSuggestedFont(font)}
```

Comment:
```suggestion
          onChange={setSuggestedFont}
```

Could this be simplier?